2024 카카오 윈터 인턴십/주사위 고르기/PGS_주사위고르기_이지영.java

```diff
@@ -0,0 +1,118 @@
+import java.util.*;
+
+public class PGS_주사위고르기_이지영 {
+
+    class Solution {
+
+        int N;
+        List<boolean[]> diceCombination;
+
+        int maxResult = 0;
+        List<Integer> answer;
+
+        int[][] dice;
+
+        public int[] solution(int[][] diceArr) {
+
+            init(diceArr);
+
+            pick(0, 0, new boolean[N]);
+
+            for (int i=0; i<diceCombination.size()/2; i++) {
+                calculateAllCase(diceCombination.get(i));
+            }
+
+            return answer.stream()
+                    .mapToInt(i->(i+1))
+                    .toArray();
+        }
+
+        void init(int[][] diceArr) {
+            N = diceArr.length;
+            diceCombination = new ArrayList<>();
+            dice = diceArr;
+        }
+
+        void pick(int now, int cnt, boolean[] picked) {
+            if(cnt==N/2) {
+                diceCombination.add(picked.clone());
+                return;
+            }
+
+            if(now==N) return;
+
+            picked[now] = true;
+            pick(now+1, cnt+1, picked);
+
+            picked[now] = false;
+            pick(now+1, cnt, picked);
```

Comment:
```suggestion
            for(int i =now;i<N;i++){
                picked[i] = true;
                pick(i+1, cnt+1, picked);
                picked[i] = false;
            }
```
A가 선택한 것만 재귀를 태우도록 이렇게 작성 할 수도 있어요!

Reply:
이쪽이 재귀 호출 depth는 더 낮겠네요! 바로 반영했어요~!